apps/asap-server/src/utils/sync-google-calendar.ts

```diff
@@ -0,0 +1,67 @@
+import { google, calendar_v3 as calendarV3, Auth } from 'googleapis';
+import logger from './logger';
+
+export type SyncCalendarFactory = (
+  syncToken: string | undefined,
+  syncEvent: (event: calendarV3.Schema$Event) => Promise<void>,
+  auth: Auth.GoogleAuth | Auth.OAuth2Client,
+) => (googleCalendarId: string) => Promise<string | undefined | null>;
+
+export const syncCalendarFactory: SyncCalendarFactory = (
+  syncToken: string | undefined,
```

Comment:
You are declaring a type right above function definition where it is used. I would suggest removing these types from the definition because they are unnecessary. 

However - if I remember correctly - there was a rule somewhere that forced that, something about being explicit, except in this scenario it is way to explicit. I tried removing those on my local and I don't see any errors but i think it might come up at build time in some check. Give that a go, that is, if you think this is a valid point. Personally I dont like clutter.

If we do have such rule and we want to leave it then the other thing you could do is declare the type like:
```typescript
export type SyncCalendarFactory = typeof syncCalendarFactory
```
Because you only really need this type so that the factory can be used as a dependency